.github/workflows/linter.yml

```diff
@@ -41,4 +41,6 @@ jobs:
       - name: Run mypy
         run: |
           pip install mypy
-          mypy .
+          pip install -r python/requirements.txt
+          mypy --config-file=.github/linters/mypy.ini run_ingestion/ python/ingestion/
```

Comment:
I would do
```
pip install -r run_ingestion/requirements.txt
pip install -r run_gcs_to_bq/requirements.txt
```

The python/requirements.txt file is just for the cloud function version so we'll get rid of it when we migrate over to the main repo. Each of the cloud run directories should have its own requirements file with everything needed for that service.

Reply:
I tried running them all separately, but because of the way mypy follows imports it makes more sense to do it this way. Basically when checking e.g. run_ingestion/, it tries to find the ingestion package but can't because of the way we structure our directories. The way around this is to tell it where the ingestion package is on the command line. It recursively checks all package directories, so when it runs either of these lines it will check everything in python/ingestion anyway. 

As for --disallow-untyped-defs, I agree. IMO we should do that sooner rather than later otherwise it will become very unwieldy.

Reply:
Yeah definitely let's add this sooner rather than later so we don't have to spend tons of time adding types to a large codebase.

If you specify on a per-package basis, then if run_ingestion adds a dep on `python/<new_package>`, would this fail or would it silently succeed but just not type check `<new_package>`? Just want to make sure this can't silently fail to type check future deps. If that doesn't work, you could change it to do the whole `python/` directory I think?

Reply:
It would fail the check and we'd have to add <new_package> to the command. I can't use the whole python/ directory now because they both define main.py modules and it won't allow duplicate modules. Once we get rid of the cloud function leftovers we could probably change it to python/.

Reply:
Ah okay on worries then. I don't think updating it to add <new_package> is a big deal, just didn't want it silently failing